### PR TITLE
Removing testing modules from prod code.

### PR DESCRIPTION
**Pull request by jsirois**
Although these were harmless, removing them was made natural with the
introduction of a dedicated test runner script which will be expanded to
include spinning up a `devpi-server` in coming work to bring down
integration test times.

### CODE REVIEW

File: tests/integration/test_interpreter_selection.py [hunk 1]
```diff
@@ -397,109 +397,108 @@ def test_pex_python():
...
-                    """
-                    from setuptools import setup
-
-                    setup(
-                        name='tester',
-                        version='1.0',
-                        description='tests',
-                        author='tester',
-                        author_email='test@test.com',
-                        packages=['testing'],
```
Comment (jsirois) [inline tests/integration/test_interpreter_selection.py:442]: I indulged a bit here with more changes than needed. The issue that needed to be fixed was just re-naming `testing` to `tester` (The `testing` package is now a top-level package in the Pex repo and clashed leading to test failure).

While reading all this I noticed it was a bit of a horror show from the Twitter days. It still is but is a bit cleaner now.

File: pex/pex_builder.py [hunk 2]
```diff
@@ -42,6 +42,13 @@
     from typing import Dict, Optional
 
 
+# N.B.: __file__ will be relative when this module is loaded from a "" `sys.path` entry under
```
Comment (jsirois) [inline pex/pex_builder.py:45]: This took me a good bit to understand and figure out but hopefully is clear now with this encoding of the fix.

The PR should be all good to review / go green now.

File: pex/resolve/resolver_configuration.py [hunk 3]
```diff
@@ -24,7 +25,7 @@
     from pex.third_party import attr
 
 
-PYPI = "https://pypi.org/simple"
+PYPI = os.environ.get("_PEX_TEST_DEFAULT_INDEX", "https://pypi.org/simple")
```
Comment (jsirois) [inline pex/resolve/resolver_configuration.py:28]: Oops - cruft from the devpi-server experiment.

**Review by benjyw — APPROVED** [review]
Sorry for the delay, took me a while to plow through this.

**Pull request merged**